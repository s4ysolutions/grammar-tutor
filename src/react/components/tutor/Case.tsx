/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {useMemo} from 'react';
import log from '../../../log';
import {Typography, useTheme} from '@mui/material';

const topSpace = 2;

const Case: React.FunctionComponent<{caseTitle: string}> = ({caseTitle}): React.ReactElement => {
  log.render('Case', caseTitle);
  const theme = useTheme();
  const sx = useMemo(() => ({
    mt: theme.spacing(topSpace),
  }), [theme]);
  return <Typography align="center" sx={sx} variant="h6">
    {caseTitle}
  </Typography>;
};

export default Case;
