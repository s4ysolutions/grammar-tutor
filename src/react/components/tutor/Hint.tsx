/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import log from '../../../log';
import React, {ReactNode, useMemo} from 'react';
import {Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, useTheme} from '@mui/material';

const upper = {'textTransform': 'uppercase', 'fontWeight': 'bold'};

const topSpace = 3;

const Hint: React.FunctionComponent<{ columnTitles: string[], children: ReactNode[]}> =
  ({columnTitles, children}): React.ReactElement => {
    log.render('Hint');

    const theme = useTheme();
    const sx = useMemo(() => ({
      mt: theme.spacing(topSpace),
      mb: theme.spacing(topSpace),
    }), [theme]);

    // return cases === null ? null : <TableContainer component={Paper} sx={sx} >
    return <TableContainer component={Paper} sx={sx} >
      <Table aria-label="simple table" >
        <TableHead >
          <TableRow >
            <TableCell />

            {
              columnTitles.map(title => <TableCell key={title} sx={upper} >
                {title}
              </TableCell >)
            }
          </TableRow >
        </TableHead >

        <TableBody >
          {children}
        </TableBody >
      </Table >
    </TableContainer >;
  };

export default Hint;
