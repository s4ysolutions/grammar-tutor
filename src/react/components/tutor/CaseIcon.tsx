/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import {Case, GrammarAnimation, GrammarGender, GrammarPlurality} from '../../../tutor';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faCloud, faPerson as faMan, faTree, faPersonDress as faWoman} from '@fortawesome/free-solid-svg-icons';

export const CSS_SHIFT_LEFT = {position: 'relative', marginLeft: -10, opacity: 0.7} as React.CSSProperties;

const CaseIcon: React.FunctionComponent<{exerciseCase: Case, plural?: boolean}> = ({exerciseCase, plural}): React.ReactElement | null => {
  const style = plural ? CSS_SHIFT_LEFT : {};
  // has gender return man/woman/cloud
  if (exerciseCase.gender === GrammarGender.NEUTER) {
    return <FontAwesomeIcon icon={faCloud} style={style} />;
  } else if (exerciseCase.gender === GrammarGender.FEMININE) {
    return <FontAwesomeIcon icon={faWoman} style={style} />;
  } else if (exerciseCase.gender === GrammarGender.MASCULINE) {
    return <FontAwesomeIcon icon={faMan} style={style} />;
  }
  // has plurality return man/tree
  if (exerciseCase.plurality === GrammarPlurality.PLURAL || exerciseCase.plurality === GrammarPlurality.SINGULAR) {
    if (exerciseCase.animation === GrammarAnimation.INANIMATE) {
      return <FontAwesomeIcon icon={faTree} style={style} />;
    }
    return <FontAwesomeIcon icon={faMan} style={style} />;
  }
  // has animation return man/tree
  if (exerciseCase.animation === GrammarAnimation.ANIMATE) {
    return <FontAwesomeIcon icon={faMan} style={style} />;
  }
  if (exerciseCase.animation === GrammarAnimation.INANIMATE) {
    return <FontAwesomeIcon icon={faTree} style={style} />;
  }
  return null;
};

export default CaseIcon;
