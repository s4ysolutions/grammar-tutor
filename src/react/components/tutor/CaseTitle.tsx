/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {ReactNode, useMemo} from 'react';
import {Typography, useTheme} from '@mui/material';
import T from '../../../l10n';
import {Case} from '../../../tutor';

const topSpace = 2;


const getTitle = (exerciseCase: Case): string => {
  const res: string[] = [T`${exerciseCase.case}`];
  if (exerciseCase.plurality !== undefined) {
    res.push(T`${exerciseCase.plurality}`);
  }
  if (exerciseCase.gender !== undefined) {
    res.push(T`${exerciseCase.gender}`);
  }
  if (exerciseCase.animation !== undefined) {
    res.push(T`${exerciseCase.animation}`);
  }
  if (exerciseCase.form !== undefined) {
    res.push(T`${exerciseCase.form}`);
  }

  return res.join(', ');
};

const CaseTitle: React.FunctionComponent<{exerciseCase: Case, children?: ReactNode[] | ReactNode}> =
  ({exerciseCase, children}): React.ReactElement => {
    const theme = useTheme();
    const sx = useMemo(() => ({
      mt: theme.spacing(topSpace),
    }), [theme]);

    return <Typography align="center" sx={sx} variant="h5">
      {children}
    &nbsp;
      {getTitle(exerciseCase)}
    </Typography>;
  };

export default CaseTitle;
