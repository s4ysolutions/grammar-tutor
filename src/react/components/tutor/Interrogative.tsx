/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {useEffect, useMemo, useState} from 'react';
import {CaseExercise} from '../../../tutor';
import diFactory from '../../../di/default';
import {Typography, useTheme} from '@mui/material';

const {di} = diFactory;
const interrogativePronounsDb = di.caseInterrogativesDb;

const Interrogative: React.FunctionComponent<{exercise: CaseExercise | null}> =
  ({exercise}): React.ReactElement | null => {
    const [interrogative, setInterrogative] = useState<string[] | null>(null);

    useEffect(() => {
      if (exercise === null) {
        setInterrogative(null);
      } else {
        interrogativePronounsDb.getInterrogativeForCase(exercise.exerciseCase.case)
          .then(cs => setInterrogative(cs.length === 0 ? null : cs.map(c => c.word)));
      }
    }, [exercise]);

    const theme = useTheme();
    const sx = useMemo(() => ({
      mt: theme.spacing(1),
    }), [theme]);

    return interrogative
      ? <Typography align="center" sx={sx} variant="h6">
        {interrogative.join('; ')}
      </Typography>
      : null;
  };

export default Interrogative;
