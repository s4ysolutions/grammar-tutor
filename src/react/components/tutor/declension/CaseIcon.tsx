/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import {Case, GrammarAnimation, GrammarGender} from '../../../../tutor';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {
  faCloud,
  faVenus as faFemale,
  faMars as faMale,
  faPerson as faMan,
  faNeuter,
  faTree,
  faPersonDress as faWoman,
} from '@fortawesome/free-solid-svg-icons';
import {CSS_SHIFT_LEFT} from '../constants';

const CaseIcon: React.FunctionComponent<{exerciseCase: Case, plural?: boolean}> = ({exerciseCase, plural}): React.ReactElement | null => {
  const style = plural ? CSS_SHIFT_LEFT : {};
  if (exerciseCase.animation === undefined || exerciseCase.animation === GrammarAnimation.ANIMATE) {
    // has gender return man/woman/cloud
    if (exerciseCase.gender === GrammarGender.NEUTER) {
      return <FontAwesomeIcon icon={faCloud} style={style} />;
    } else if (exerciseCase.gender === GrammarGender.FEMININE) {
      return <FontAwesomeIcon icon={faWoman} style={style} />;
    } else if (exerciseCase.gender === GrammarGender.MASCULINE) {
      return <FontAwesomeIcon icon={faMan} style={style} />;
    }
  }

  if (exerciseCase.gender === GrammarGender.NEUTER) {
    return <FontAwesomeIcon icon={faNeuter} style={style} />;
  } else if (exerciseCase.gender === GrammarGender.FEMININE) {
    return <FontAwesomeIcon icon={faFemale} style={style} />;
  } else if (exerciseCase.gender === GrammarGender.MASCULINE) {
    return <FontAwesomeIcon icon={faMale} style={style} />;
  }

  // has plurality return man/tree
  if (exerciseCase.plurality !== undefined) {
    return <FontAwesomeIcon icon={faMan} style={style} />;
  }

  // no gender
  if (exerciseCase.animation === GrammarAnimation.ANIMATE) {
    return <FontAwesomeIcon icon={faMan} style={style} />;
  }
  if (exerciseCase.animation === GrammarAnimation.INANIMATE) {
    return <FontAwesomeIcon icon={faTree} style={style} />;
  }

  return null;
};

export default CaseIcon;
