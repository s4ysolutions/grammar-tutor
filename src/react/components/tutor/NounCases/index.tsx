/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {useCallback, useMemo, useState} from 'react';
import {Container, IconButton} from '@mui/material';
import usePromise from '../../../hooks/usePromise';
import {NounCaseExercise} from '../../../../tutor';
import Noun from './Noun';
import Variants from './Variants';
import log from '../../../../log';
import Case from './Case';
import {getDi} from '../../../../di/default';
import Hint from './Hint';
import QuizIcon from '@mui/icons-material/Quiz';
import T from '../../../../l10n';
import Grid2 from '@mui/material/Unstable_Grid2';

const di = getDi();
const tutor = di.tutor;
const exerciseIssuer = tutor.nextPronounExersizeSelectWord.bind(tutor);

const NounCases: React.FunctionComponent = (): React.ReactElement => {
  log.render('NounCases');

  const [currentExercise, nextExercise] =
    usePromise<NounCaseExercise | null>(exerciseIssuer, null, 'NounCases');

  const [help, setHelp] = useState(false);

  const toggleHelp = useCallback(() => setHelp(!help), [help, setHelp]);

  const possibleVariant = useMemo(() =>
    currentExercise === null ? null : currentExercise.possibleVariants.shuffle(), [currentExercise]);
  return <Container >
    {currentExercise ? <React.Fragment >
      <Noun noun={currentExercise.mainForm} />

      <Case exerciseCase={currentExercise.exerciseCase} />

      <Variants
        exercise={currentExercise}
        nextExercise={nextExercise}
        possibleVariants={possibleVariant} />

      <Grid2 container justifyContent="right">
        <IconButton aria-label={T`Hint`} color="primary" onClick={toggleHelp}>
          <QuizIcon />
        </IconButton >
      </Grid2>

      {help ? <Hint exerciseCase={currentExercise.exerciseCase} word={currentExercise.mainForm} /> : null}
    </React.Fragment >
      : null}

  </Container >;
};

export default NounCases;
