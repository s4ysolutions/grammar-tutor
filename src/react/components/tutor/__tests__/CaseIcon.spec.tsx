/*
 * Copyright 2022 by s4y.solutions
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import diFactory, {DefaultDi} from '../../../../di/default';
import {Di} from '../../../../di';
import memoryStoragePromiseFactory from '../../../../../tests/mocks/kv-promice/memoryStorage';
import renderer from 'react-test-renderer';
import CaseIcon from '../CaseIcon';
import {Case, GrammarAnimation, GrammarCase, GrammarGender, GrammarPlurality} from '../../../../tutor';

describe('<CaseIcon>', () => {

  beforeEach(() => {
    (diFactory as { di: Di }).di = new DefaultDi(memoryStoragePromiseFactory({}));
  });

  describe('Empty', () => {
    test('no props should return null', () => {
      const c: Case = {
        case: GrammarCase.NOMINATIVE,
        word: 'foo',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).toBeNull();
      // expect(tree).toMatchSnapshot();
      // expect(tree).toHaveProperty('props');
    });
  });

  describe('Nouns with animation (case interrogatives)', () => {
    test('animate should return person', () => {
      const c: Case = {
        animation: GrammarAnimation.ANIMATE,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'person');
    });

    test('inanimate should return tree', () => {
      const c: Case = {
        animation: GrammarAnimation.INANIMATE,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'tree');
    });
  });

  describe('Nouns with plurality only (personal pronouns 1st and 2nd)', () => {
    test('singular should return person', () => {
      const c: Case = {
        plurality: GrammarPlurality.SINGULAR,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'person');
    });

    test('plural should return person', () => {
      const c: Case = {
        plurality: GrammarPlurality.PLURAL,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'person');
    });
  });

  describe('Nouns with plurality and gender (nouns, personal pronouns 3d)', () => {
    test('singular masculine should return person', () => {
      const c: Case = {
        gender: GrammarGender.MASCULINE,
        plurality: GrammarPlurality.SINGULAR,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'person');
    });

    test('plural masculine should return person', () => {
      const c: Case = {
        gender: GrammarGender.MASCULINE,
        plurality: GrammarPlurality.PLURAL,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'person');
    });

    test('singular feminine should return person-dress', () => {
      const c: Case = {
        gender: GrammarGender.FEMININE,
        plurality: GrammarPlurality.SINGULAR,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'person-dress');
    });

    test('plural feminine should return person-dress', () => {
      const c: Case = {
        gender: GrammarGender.FEMININE,
        plurality: GrammarPlurality.PLURAL,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'person-dress');
    });

    test('singular neuter should return cloud', () => {
      const c: Case = {
        gender: GrammarGender.NEUTER,
        plurality: GrammarPlurality.SINGULAR,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'cloud');
    });

    test('plural neuter should return cloud', () => {
      const c: Case = {
        gender: GrammarGender.NEUTER,
        plurality: GrammarPlurality.PLURAL,
        case: GrammarCase.NOMINATIVE,
        word: '',
      };
      const component = renderer.create(<CaseIcon exerciseCase={c} />);
      const tree = component.toJSON();
      expect(tree).not.toBeNull();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveProperty('props.data-icon', 'cloud');
    });
  });
});
