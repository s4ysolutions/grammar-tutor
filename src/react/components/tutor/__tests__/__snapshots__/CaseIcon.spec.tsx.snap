// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaseIcon> Nouns with animation (case interrogatives) animate should return person 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-person "
  data-icon="person"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 320 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M208 48c0 26.5-21.5 48-48 48s-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48zM152 352V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V256.9L59.4 304.5c-9.1 15.1-28.8 20-43.9 10.9s-20-28.8-10.9-43.9l58.3-97c17.4-28.9 48.6-46.6 82.3-46.6h29.7c33.7 0 64.9 17.7 82.3 46.6l58.3 97c9.1 15.1 4.2 34.8-10.9 43.9s-34.8 4.2-43.9-10.9L232 256.9V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V352H152z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with animation (case interrogatives) inanimate should return tree 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-tree "
  data-icon="tree"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 448 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M210.6 5.9L62 169.4c-3.9 4.2-6 9.8-6 15.5C56 197.7 66.3 208 79.1 208H104L30.6 281.4c-4.2 4.2-6.6 10-6.6 16C24 309.9 34.1 320 46.6 320H80L5.4 409.5C1.9 413.7 0 419 0 424.5c0 13 10.5 23.5 23.5 23.5H192v32c0 17.7 14.3 32 32 32s32-14.3 32-32V448H424.5c13 0 23.5-10.5 23.5-23.5c0-5.5-1.9-10.8-5.4-15L368 320h33.4c12.5 0 22.6-10.1 22.6-22.6c0-6-2.4-11.8-6.6-16L344 208h24.9c12.7 0 23.1-10.3 23.1-23.1c0-5.7-2.1-11.3-6-15.5L237.4 5.9C234 2.1 229.1 0 224 0s-10 2.1-13.4 5.9z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality and gender (nouns, personal pronouns 3d) plural femine should return person-dress 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-person-dress "
  data-icon="person-dress"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 320 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M160 96c-26.5 0-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48s-21.5 48-48 48zM88 384H70.2c-10.9 0-18.6-10.7-15.2-21.1L93.3 248.1 59.4 304.5c-9.1 15.1-28.8 20-43.9 10.9s-20-28.8-10.9-43.9l53.6-89.2c20.3-33.7 56.7-54.3 96-54.3h11.6c39.3 0 75.7 20.6 96 54.3l53.6 89.2c9.1 15.1 4.2 34.8-10.9 43.9s-34.8 4.2-43.9-10.9l-33.9-56.3L265 362.9c3.5 10.4-4.3 21.1-15.2 21.1H232v96c0 17.7-14.3 32-32 32s-32-14.3-32-32V384H152v96c0 17.7-14.3 32-32 32s-32-14.3-32-32V384z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality and gender (nouns, personal pronouns 3d) plural masculine should return person 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-person "
  data-icon="person"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 320 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M208 48c0 26.5-21.5 48-48 48s-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48zM152 352V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V256.9L59.4 304.5c-9.1 15.1-28.8 20-43.9 10.9s-20-28.8-10.9-43.9l58.3-97c17.4-28.9 48.6-46.6 82.3-46.6h29.7c33.7 0 64.9 17.7 82.3 46.6l58.3 97c9.1 15.1 4.2 34.8-10.9 43.9s-34.8 4.2-43.9-10.9L232 256.9V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V352H152z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality and gender (nouns, personal pronouns 3d) plural neuter should return cloud 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-cloud "
  data-icon="cloud"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 640 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 336c0 79.5 64.5 144 144 144H512c70.7 0 128-57.3 128-128c0-61.9-44-113.6-102.4-125.4c4.1-10.7 6.4-22.4 6.4-34.6c0-53-43-96-96-96c-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32C167.6 32 96 103.6 96 192c0 2.7 .1 5.4 .2 8.1C40.2 219.8 0 273.2 0 336z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality and gender (nouns, personal pronouns 3d) singular femine should return person-dress 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-person-dress "
  data-icon="person-dress"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 320 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M160 96c-26.5 0-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48s-21.5 48-48 48zM88 384H70.2c-10.9 0-18.6-10.7-15.2-21.1L93.3 248.1 59.4 304.5c-9.1 15.1-28.8 20-43.9 10.9s-20-28.8-10.9-43.9l53.6-89.2c20.3-33.7 56.7-54.3 96-54.3h11.6c39.3 0 75.7 20.6 96 54.3l53.6 89.2c9.1 15.1 4.2 34.8-10.9 43.9s-34.8 4.2-43.9-10.9l-33.9-56.3L265 362.9c3.5 10.4-4.3 21.1-15.2 21.1H232v96c0 17.7-14.3 32-32 32s-32-14.3-32-32V384H152v96c0 17.7-14.3 32-32 32s-32-14.3-32-32V384z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality and gender (nouns, personal pronouns 3d) singular masculine should return person 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-person "
  data-icon="person"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 320 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M208 48c0 26.5-21.5 48-48 48s-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48zM152 352V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V256.9L59.4 304.5c-9.1 15.1-28.8 20-43.9 10.9s-20-28.8-10.9-43.9l58.3-97c17.4-28.9 48.6-46.6 82.3-46.6h29.7c33.7 0 64.9 17.7 82.3 46.6l58.3 97c9.1 15.1 4.2 34.8-10.9 43.9s-34.8 4.2-43.9-10.9L232 256.9V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V352H152z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality and gender (nouns, personal pronouns 3d) singular neuter should return cloud 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-cloud "
  data-icon="cloud"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 640 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 336c0 79.5 64.5 144 144 144H512c70.7 0 128-57.3 128-128c0-61.9-44-113.6-102.4-125.4c4.1-10.7 6.4-22.4 6.4-34.6c0-53-43-96-96-96c-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32C167.6 32 96 103.6 96 192c0 2.7 .1 5.4 .2 8.1C40.2 219.8 0 273.2 0 336z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality only (personal pronouns 1st and 2nd) plural should return person 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-person "
  data-icon="person"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 320 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M208 48c0 26.5-21.5 48-48 48s-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48zM152 352V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V256.9L59.4 304.5c-9.1 15.1-28.8 20-43.9 10.9s-20-28.8-10.9-43.9l58.3-97c17.4-28.9 48.6-46.6 82.3-46.6h29.7c33.7 0 64.9 17.7 82.3 46.6l58.3 97c9.1 15.1 4.2 34.8-10.9 43.9s-34.8 4.2-43.9-10.9L232 256.9V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V352H152z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;

exports[`<CaseIcon> Nouns with plurality only (personal pronouns 1st and 2nd) singular should return person 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-person "
  data-icon="person"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={{}}
  viewBox="0 0 320 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M208 48c0 26.5-21.5 48-48 48s-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48zM152 352V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V256.9L59.4 304.5c-9.1 15.1-28.8 20-43.9 10.9s-20-28.8-10.9-43.9l58.3-97c17.4-28.9 48.6-46.6 82.3-46.6h29.7c33.7 0 64.9 17.7 82.3 46.6l58.3 97c9.1 15.1 4.2 34.8-10.9 43.9s-34.8 4.2-43.9-10.9L232 256.9V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V352H152z"
    fill="currentColor"
    style={{}}
  />
</svg>
`;
